kola/tests/locksmith/locksmith.go

```diff
@@ -117,6 +121,41 @@ func init() {
 	})
 }
 
+func locksmithReboot(c cluster.TestCluster) {
+	// The machine should be able to reboot without etcd in the default mode
+	m := c.Machines()[0]
+
+	bootId1, err := m.SSH("cat /proc/sys/kernel/random/boot_id")
+	if err != nil {
+		c.Fatalf("failed to run \"cat /proc/sys/kernel/random/boot_id\": output: %q status: %q", bootId1, err)
+	}
+
+	output, err := m.SSH("locksmithctl send-need-reboot")
+	if _, ok := err.(*ssh.ExitMissingError); ok {
+		err = nil // A terminated session is perfectly normal during reboot.
+	} else if err == io.EOF {
+		err = nil // Sometimes copying command output returns EOF here.
+	}
+	if err != nil {
+		c.Fatalf("failed to run \"locksmithctl send-need-reboot\": output: %q status: %q", output, err)
+	}
+
+	err = platform.CheckMachine(m)
```

Comment:
This still has a race between `send-need-reboot` and `CheckMachine`.  I checked what `RebootMachine` does and it stops `sshd.socket` first to avoid races; so does `locksmithCluster` in this file.  If you make that change, the `boot_id` checks shouldn't be necessary after all.

Reply:
I updated this to change the command to `sudo systemctl stop sshd.socket && locksmithctl send-need-reboot` and removed the boot_id stuff.